### PR TITLE
[JENKINS-65493] Allow excluding by topics

### PR DESCRIPTION
**Pull request by ingwarsw**
# Description

Allow excluding topics by prepending them with `-` sign.

See [JENKINS-65493](https://issues.jenkins.io/browse/JENKINS-65493) for further information. 

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Automated tests have been added to exercise the changes
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verify that the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

# Documentation changes
- [ ] Link to jenkins.io PR, or an explanation for why no doc changes are needed

# Users/aliases to notify

### CODE REVIEW

File: src/main/resources/org/jenkinsci/plugins/github_branch_source/TopicsTrait/help-topicList.html [hunk 1]
```diff
@@ -1,3 +1,3 @@
 <div>
-    <p>Specify a comma-separated list of <strong>topics</strong> to filter for repositories that have <u>all of them</u>.</p>
-</div>
\ No newline at end of file
+    <p>Specify a comma-separated list of <strong>topics</strong> to filter for repositories that have (you can prefix topic with `-` to have it <u>excluded</u>) <u>all of them</u>.</p>
```
Comment (ingwarsw) [inline src/main/resources/org/jenkinsci/plugins/github_branch_source/TopicsTrait/help-topicList.html:2]: If someone have better idea for that help message It surely could be updated.

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java [hunk 2]
```diff
@@ -1239,9 +1212,44 @@ public void visitSources(SCMSourceObserver observer) throws IOException, Interru
         }
     }
 
+    private boolean topicMatches(final GitHubSCMNavigatorContext context, final GHRepository repo, PrintStream logger)
+            throws IOException {
+        if (context.getTopics().isEmpty()) return true;
```
Comment (timja) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java:1217]:
```suggestion
        if (context.getTopics().isEmpty()) {
            return true;
        }
```

**Review by timja — APPROVED** [review]
LGTM I'm not a maintainer here though

**Comment by timja** [comment]
@olamy / @jglick / @jtnord?

**Comment by jglick** [comment]
(I am not maintaining this plugin.)

**Comment by timja** [comment]
> (I am not maintaining this plugin.)

you're in the maintainers team FWIW https://github.com/orgs/jenkinsci/teams/github-branch-source-plugin-developers

**Pull request merged**